### DISCUSSION TITLE
Hyas & Tailwind CSS

### DISCUSSION THREAD

**Discussion by pensivedog** [gethyas/hyas#413] [Q&A] [answered]
Has anybody successfully set up Hyas to work with Tailwind CSS? I'd love to incorporate the dev tools and other elements from [hugo-tailwindcss-starter-theme](https://github.com/dirkolbrich/hugo-tailwindcss-starter-theme), but this calls for setting up a new package.json in the theme folder with a lot of overlap with packages already installed in Hyas.

If anyone could provide steps on how they made this or something similar work, I'll be very grateful!

**Comment by h-enk** [comment]
Just started working on this – you triggered me earlier 😉

It should be fairly simple now (with the new Hyas setup, the just released v1.0.3 of `create-hyas`, and the great setup of Dirk). 

Will get back to you!

**Reply by h-enk** [reply]
Got it working /set up 🚀  Just making it a little more robust and running some more tests before releasing it.

**Reply by pensivedog** [reply]
Amazing! Can't wait to see the release :)

**Reply by h-enk** [reply, marked as answer]
OK, I finally found the time to finish this 🎉 

- Official starter: [tailwindcss-starter](https://github.com/gethyas/tailwindcss-starter)
- Official integration: [tailwindcss](https://github.com/gethyas/tailwindcss)

Included are the latest versions of:
- [tailwindcss](https://github.com/tailwindlabs/tailwindcss)
- [@tailwindcss/aspect-ratio](https://github.com/tailwindlabs/tailwindcss-aspect-ratio)
- [@tailwindcss/forms](https://github.com/tailwindlabs/tailwindcss-forms)
- [@tailwindcss/typography](https://github.com/tailwindlabs/tailwindcss-typography)

Note that [@tailwindcss/line-clamp](https://github.com/tailwindlabs/tailwindcss-line-clamp) is included in [tailwindcss](https://github.com/tailwindlabs/tailwindcss)

I mainly based it on [bep/hugo-starter-tailwind-basic](https://github.com/bep/hugo-starter-tailwind-basic) but also took note of [dirkolbrich/hugo-tailwindcss-starter-theme](https://github.com/dirkolbrich/hugo-tailwindcss-starter-theme).

I kept the starter basic (so not adding the dev tools and other stuff from the latter repo) to serve a wider range of use cases. If you'd like to, you could include the dev tools and other stuff in `./themes/my-tailwindcss-theme/layouts/`

Further documentation is in the making.

**Reply by pensivedog** [reply]
Wow, this is wonderful! Just tried and it seems to work like a charm. Thank you again for your brilliant work on this project!

**Reply by pensivedog** [reply]
Trying to get DaisyUI working with the Hyas/Tailwind setup. Following the setup instructions for use as a [Tailwind plugin](https://daisyui.com/docs/install/) doesn't work, although it works fine from the CDN. 

Is there some additional step needed to make Tailwind plugins like this work with Hyas?

**Reply by h-enk** [reply]
Did you add a corresponding mount? See also https://github.com/gethyas/tailwindcss#setup

**Reply by pensivedog** [reply]
I tried but I think I'm missing something. The following mounts aren't working:

 `module.toml`: 
```
[[mounts]]
  source = "node_modules/daisyui/dist"
  target = "assets/css"

[[mounts]]
  source = "node_modules/daisyui/dist"
  target = "assets/js"
```

Is this not the correct approach?

**Reply by h-enk** [reply]
What about:

```toml
[[mounts]]
  source = "node_modules/daisyui/dist"
  target = "assets"
```

**Reply by pensivedog** [reply]
As soon as `require('daisyui')` is added to the plugins in `tailwind.config.js` all site styling breaks.

**Reply by h-enk** [reply]
Are you using the right order, see notification: https://daisyui.com/docs/layout-and-typography/#-1

**Reply by h-enk** [reply]
I can have a closer look coming Wednesday (when I'm back home)

**Reply by pensivedog** [reply]
Yes, I'm using the order specified there. I'm also testing this on a clean install of Hyas with the Tailwind starter, which is having the same issue.

Appreciate you looking into it!
